Improve docstrings in Metadata API to be more descriptive

ivanayov opened the pull request:
This change updates some parts of the Metadata API docstrings
that did not give enough details and context

Fixes #1600

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [n/a] Tests have been added for the bug fix or new feature
- [n/a] Docs have been added for the bug fix or new feature

ivanayov committed:
Improve docstrings in Metadata API to be more descritpive

This change updates some parts of the Metadata API docstrings
that did not give enough details and context

Fixes #1600

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

lukpueh commented on tuf/api/metadata.py:
```diff
@@ -307,7 +307,8 @@ def sign(
 
         Args:
-            signer: A securesystemslib.signer.Signer implementation.
+            signer: Custom or securesystemslib implementation of the
+                ``securesystemslib.signer.Signer`` interface.
```
This change does not seem like a huge improvement wrt https://github.com/theupdateframework/python-tuf/pull/1815#pullrequestreview-875726152.

What about
```suggestion
            signer: A ``securesystemslib.signer.Signer`` object that provides a private
                key and signing implementation to generate the signature. A standard
                implementation is available in ``securesystemslib.signer.SSlibSigner``.
```

lukpueh reviewed (commented):
Thanks for the updates, @ivanayov! I am unsure if the "dict -> json/dict" changes really add much clarification, but I don't object.

I do have an alternative suggestion for the `Signer` docstring though. Let me know what you think.

ivanayov committed:
Improve Signer docstrings in Metadata API

Change to @lukpueh proposal with more clarification on why and how
the `securesystemslib.signer.Signer` interface is used

Co-authored-by: Lukas Pühringer <luk.puehringer@gmail.com>

Signed-off-by: Ivana Atanasova <iyovcheva@vmware.com>

ivanayov commented:
Thank you @lukpueh , I updated with your proposal.

lukpueh reviewed (approved):
Thanks!
